### PR TITLE
Feat: vesting contract

### PR DESCRIPTION
**Pull request by swr1bm86**
Fixes https://github.com/Developer-DAO/code-claim-site/issues/84

### CODE REVIEW

**Comment by swr1bm86** [comment]
```
-------------------------|----------|----------|----------|----------|----------------|
File                     |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
-------------------------|----------|----------|----------|----------|----------------|
 src/                    |      100 |    79.41 |      100 |      100 |                |
  CODE.sol               |      100 |      100 |      100 |      100 |                |
  ClaimCODE.sol          |      100 |    91.67 |      100 |      100 |                |
  MerkleProof.sol        |      100 |      100 |      100 |      100 |                |
  Vesting.sol            |      100 |       70 |      100 |      100 |                |
 src/deprecated/         |        0 |        0 |        0 |        0 |                |
  CODEToken.sol          |        0 |        0 |        0 |        0 |... 62,66,70,71 |
 src/mocks/              |      100 |      100 |      100 |      100 |                |
  MerkleProofWrapper.sol |      100 |      100 |      100 |      100 |                |
-------------------------|----------|----------|----------|----------|----------------|
All files                |    74.71 |     58.7 |    73.08 |    74.71 |                |
-------------------------|----------|----------|----------|----------|----------------|
```

File: packages/hardhat/src/Vesting.sol [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        uint256 timestamp = block.timestamp;
+        if (timestamp > start + duration) {
+            return shares[_account] - released[_account];
+        } else {
+            uint256 _epoch = epoch(timestamp - start);
+            return (shares[_account] * _epoch) / totalEpochs - released[_account];
+        }
+    }
+
+    function addPayees(address[] memory _payees, uint256[] memory _shares) public {
```
Comment (parseb) [inline packages/hardhat/src/Vesting.sol:52]: onlyOwner maybe? what am I missing?

File: packages/hardhat/src/Vesting.sol [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    // The Founding Team will retain 6%, current Advisors will retain 0.9% and only 50% will be vested
+    uint256 public immutable totalShares = (690_000 / 2) * 1e18;
+
+    constructor(address _codeToken, uint256 _startTimestamp) {
+        codeToken = IERC20(_codeToken);
+        start = _startTimestamp;
+    }
+
+    function release() public {
+        address account = _msgSender();
```
Comment (parseb) [inline packages/hardhat/src/Vesting.sol:31]: I would use msg.sender

**Review by parseb — CHANGES_REQUESTED** [review]

File: packages/hardhat/src/Vesting.sol (hunk 2 above)
Comment (swr1bm86) [reply packages/hardhat/src/Vesting.sol:31]: for reducing gas cost maybe?

File: packages/hardhat/src/Vesting.sol (hunk 1 above)
Comment (swr1bm86) [reply packages/hardhat/src/Vesting.sol:52]: correct, will fix that!

**Pull request closed without merging**